rc/windowing/detection.kak

```diff
@@ -0,0 +1,63 @@
+# Attempt to detect the windowing environment we're operating in
+#
+# We try to load modules from the windowing_modules str-list option in order,
+# stopping when one of the modules loads successfully. This ensures that only
+# a single module is loaded by default.
+#
+# On load each module must attempt to detect the environment it's appropriate
+# for, and if the environment isn't appropriate it must fail with an error.
+# In addition, each module must check for the length of the windowing_modules
+# str-list option defined below, and must /not/ check for an appropriate
+# environment if the list is empty. An example of this test:
+#
+# evaluate-commands %sh{
+#     set -- ${kak_opt_windowing_modules}
+#     [ $# -eq 0 ] || [ -n "$TMUX" ] || echo 'fail tmux not detected'
+# }
+#
+# To override the default detection ordering add something like the following
+# to your kakrc:
+#
+# set-option global windowing_modules 'mod1' 'mod2' 'mod3'
+#
+# To disable the whole process, set the windowing_modules option to an empty
+# list - manually loading the modules will still work, with the environment
+# detection disabled.
+#
+# Each module is expected to define at least two aliases:
+#  * terminal - create a new terminal with sensible defaults
+#  * focus - focus the specified client, defaulting to the current client
+#
+
+declare-option -docstring \
+"Ordered list of windowing modules to try and load.
+
+Default: tmux screen kitty iterm x11
+
+To override the default add a line like this to your kakrc:
+
+ set-option global windowing_modules 'mod1' 'mod2' 'mod3'
+
+To disble automatic loading of windowing modules so you can
+customise the module loading yourself set the windowing_modules
+option to an empty list:
+
+ set-option global windowing_modules
+
+and add your own module loading code." \
```

Comment:
I think this docstring goes a bit too much into details. I dont think options should document their default value and what syntax is used to set them in the docstring, current value is displayed as a completion, and how set-option work is general Kakoune knowledge.

Reply:
Trying to be self-documenting and discoverable, I guess that docstring went way overboard . . .